src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/InProcDataCollectionExtensionManager.cs

```diff
@@ -61,13 +61,13 @@ protected InProcDataCollectionExtensionManager(string? runSettings, ITestEventsP
         _inProcDataCollectionSink = new InProcDataCollectionSink();
         _defaultCodeBase = defaultCodeBase;
         _fileHelper = fileHelper;
-        _codeBasePaths = new List<string?> { _defaultCodeBase };
+        CodeBasePaths = new HashSet<string?> { _defaultCodeBase };
```

Comment:
I would recommend to use the constructor of `HashSet` that takes the comparer. If you add another test where 2 paths are the same except for the casing, you will see that you will not get the deduplication.